### PR TITLE
Sscs 3265 evidence upload

### PR DESCRIPTION
**Pull request by barbaracassani**
This is very basic. Includes the feature flag and a fake api to accept the forwarding of the uploaded files.

### CODE REVIEW

File: paths.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
+const _ = require('lodash');
+
```
Comment (paulgain) [inline paths.js]: Can we use the destructuring assignment please: `const { get } = require('lodash');`

File: steps.js [hunk 2]
```diff
@@ -1,4 +1,8 @@
 /* eslint-disable max-len */
+const _ = require('lodash');
+
+const evidenceUploadEnabled = _.get(require('config'), 'features.evidenceUpload.enabled');
+
```
Comment (paulgain) [inline steps.js]: Same applies here, we need to be consistent. Thanks.

File: steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+const fs = require('fs');
+const request = require('request');
+
+class EvidenceUpload extends Question {
+  static get path() {
+    return paths.reasonsForAppealing.evidenceUpload;
+  }
+
+  static handleUpload(req, res, next) {
+    const pathToUploadFolder = './../../../uploads';
```
Comment (paulgain) [inline steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js]: Isn't it convention to use __dir when dealing with paths on Node/Express? For example:
`const pathToUploadFolder = '__dir/path/to/uploads'`

File: steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
+const { Question, goTo } = require('@hmcts/one-per-page');
+const { form, text } = require('@hmcts/one-per-page/forms');
+const { api } = require('../../../config/default');
```
Comment (paulgain) [inline steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js]: For consistency please use the full path, no relative paths.

File: steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+const { Logger } = require('@hmcts/nodejs-logging');
```
Comment (paulgain) [inline steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js]: Please group all dependencies such as `@hmcts/dependency` together.

File: steps/reasons-for-appealing/sending-evidence/SendingEvidence.js [hunk 6]
```diff
@@ -2,6 +2,9 @@ const { Question, goTo } = require('@hmcts/one-per-page');
 const { form, text } = require('@hmcts/one-per-page/forms');
 const { answer } = require('@hmcts/one-per-page/checkYourAnswers');
 const paths = require('paths');
+const _ = require('lodash');
```
Comment (paulgain) [inline steps/reasons-for-appealing/sending-evidence/SendingEvidence.js]: Same applies here.

File: paths.js (hunk 1 above)
Comment (barbaracassani) [reply paths.js]: Happy to do that, but what is exactly the rationale behind it? Just curious.

File: steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js (hunk 3 above)
Comment (barbaracassani) [reply steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js]: Is it? Aren't you thinking to `__dirname` maybe?

File: paths.js (hunk 1 above)
Comment (paulgain) [reply paths.js]: Readability and consistency throughout Reform.

File: steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js (hunk 3 above)
Comment (paulgain) [reply steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js]: Doh, sorry yes, `__dirname`

Comment (barbaracassani) [reply steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js]: Anyway, I am resolving that later... but thanks to your comment I've discovered I had assigned that to a `const` beforehand, so I will remove that.

File: steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
+const { Question, goTo } = require('@hmcts/one-per-page');
+const { form, text } = require('@hmcts/one-per-page/forms');
+const { Logger } = require('@hmcts/nodejs-logging');
+const { api } = require('config/default.json');
```
Comment (paulgain) [inline steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js]: Config does not work like this, there's no need to explicitly define `default.json` or `custom-environment-variables.json` or any other environment as part of the require. Config does this for you based on the NODE_ENV environment variable.

Locally
```
const config = require('config');
const url = config.get('api.url'); // http://localhost:8080
const uploadEvidenceUrl = config.get('api.uploadEvidenceUrl'); // http://localhost:3010/upload
```
NODE_ENV=aat or prod 
```
const config = require('config');
const url = config.get('api.url'); // Comes from the env. var. TRIBUNALS_CASE_API_URL 
const uploadEvidenceUrl = config.get('api.uploadEvidenceUrl'); // This should come from an environment variable (you haven't created this)
```

File: config/default.json [hunk 8]
```diff
@@ -28,5 +29,10 @@
     "waitForAction": "500",
     "outputDir": "./output",
     "frontendUrl": "http://localhost:3000"
+  },
+  "features": {
+    "evidenceUpload": {
+      "enabled": false
+    }
   }
```
Comment (bendiggle) [inline config/default.json]: These need to be added to custom-environment-variables.json too

**Pull request merged**